packages/hardhat-core/src/internal/hardhat-network/provider/modules/logger.ts

```diff
@@ -1,12 +1,18 @@
 import util from "util";
 
 export class ModulesLogger {
-  public enabled = false;
+  private _enabled = false;
   private _logs: Array<string | [string, string]> = [];
   private _titleLength = 0;
+  private _indentEnabled = false;
+  private _indent = 4;
 
-  public enable(isEnabled: boolean) {
-    this.enabled = isEnabled;
+  public get enabled() {
+    return this._enabled;
+  }
+
+  public setEnabled(enabled: boolean) {
+    this._enabled = enabled;
   }
```

Comment:
Mixing two styles of get/set methods is **ugly**. 
* Note that when a user sees `logger.enabled` field available on object, they expect `logger.enabled = true` to work as well. This is not the case when there's no Typescript `set` method.
* Typescript get/set methods were added to provide compatibility with JavaScript in which you can effectively **overload** the assignment and retrieval of a field. This is probably a matter of personal preference, but unless you're writing types for a legacy JS library, I'd stick with using normal functions for getters and setters - they indicate more that there might be some logic underneath than normal retrieval / assignment.
```suggestion
  public isEnabled() {
    return this._enabled;
  }

  public setEnabled(enabled: boolean) {
    this._enabled = enabled;
  }
```